## Jupyter notebook index

<pre>
1. <b>Dataset</b>
    1.1 Information about our dataset
          -- Script to download data from google drive
    1.2 Information about benchmark dataset
          -- Script to download data from google drive

2. <b>Pickles file</b>
      Information about all the pickles files containing embedding vectors and sequences
           --- Script to download pickles file

3. <b>Model Weights and architectures</b>
      Script to download trained model weights and model architecture(another jupyter notebook)
           -
</pre>

In [1]:
import pandas as pd

## 1. Information about Dataset

<pre>
<b>Link to get all the required files to repeat the experiment:</b> https://drive.google.com/drive/folders/1bi1_9A5YOUtBRBuO5GRZEaPcsXGbUPOP?usp=share_link

<b>Raw dataset from published articles and databases: </b>https://drive.google.com/drive/folders/17d7w3JpKmenS-iN5y_arOOj-OS3FI16t?usp=share_link
</pre>


## 1.1 Our Dataset



In [3]:
!wget --no-check-certificate 'https://docs.google.com/uc?export=download&id=1wWO-I8cExssypW_HFOu6xaw0UYM4shlG' -O train_set.csv
!wget --no-check-certificate 'https://docs.google.com/uc?export=download&id=1VzkSzZaTsM86Yp9gLLRXsWublp5zQdAZ' -O test_set.csv

--2023-01-23 05:45:36--  https://docs.google.com/uc?export=download&id=1wWO-I8cExssypW_HFOu6xaw0UYM4shlG
Resolving docs.google.com (docs.google.com)... 142.250.101.100, 142.250.101.101, 142.250.101.138, ...
Connecting to docs.google.com (docs.google.com)|142.250.101.100|:443... connected.
HTTP request sent, awaiting response... 303 See Other
Location: https://doc-14-2s-docs.googleusercontent.com/docs/securesc/ha0ro937gcuc7l7deffksulhg5h7mbp1/honmshh7kbis3f2adaljujrntjueaer0/1674452700000/17049954329865554887/*/1wWO-I8cExssypW_HFOu6xaw0UYM4shlG?e=download&uuid=41c795ec-365f-499f-9934-99c1f3752a7d [following]
--2023-01-23 05:45:38--  https://doc-14-2s-docs.googleusercontent.com/docs/securesc/ha0ro937gcuc7l7deffksulhg5h7mbp1/honmshh7kbis3f2adaljujrntjueaer0/1674452700000/17049954329865554887/*/1wWO-I8cExssypW_HFOu6xaw0UYM4shlG?e=download&uuid=41c795ec-365f-499f-9934-99c1f3752a7d
Resolving doc-14-2s-docs.googleusercontent.com (doc-14-2s-docs.googleusercontent.com)... 142.251.2.132, 2607:f8

In [7]:
train_set = pd.read_csv('train_set.csv')
test_set = pd.read_csv('test_set.csv')

In [14]:
train_set.head(2)

Protein_A  Protein_B  label
0  PRO_1968   PRO_3259      1
1  PRO_1968  PRO_20825      1

In [15]:
test_set.head(2)

Protein_A  Protein_B  label
0   PRO_4691   PRO_2365      1
1  PRO_12486  PRO_12486      1

In [29]:
print(f"Number of positive in points train set: {len(train_set[train_set.label == 1])}")
print(f"Number of positive in points train set: {len(train_set[train_set.label == 0])}")
print(f"Number of positive in points test set: {len(test_set[test_set.label == 1])}")
print(f"Number of positive in points test set: {len(test_set[test_set.label == 0])}")

Number of positive in points train set: 241819
Number of positive in points train set: 241787
Number of positive in points test set: 21347
Number of positive in points test set: 213458


## 1.2 Benchmark Dataset

In [26]:
!wget --no-check-certificate 'https://docs.google.com/uc?export=download&id=1U1DN4XnhZmEaY7ivoPdj8tY0tau7X0sC' -O test_set_benchmark.csv
!wget --no-check-certificate 'https://docs.google.com/uc?export=download&id=1t4AXRG0uITtlqsKuRSgDVmmwsn3t2NRM' -O train_set_benchmark.csv

--2023-01-23 05:54:46--  https://docs.google.com/uc?export=download&id=1U1DN4XnhZmEaY7ivoPdj8tY0tau7X0sC
Resolving docs.google.com (docs.google.com)... 142.251.2.101, 142.251.2.100, 142.251.2.138, ...
Connecting to docs.google.com (docs.google.com)|142.251.2.101|:443... connected.
HTTP request sent, awaiting response... 303 See Other
Location: https://doc-00-2s-docs.googleusercontent.com/docs/securesc/ha0ro937gcuc7l7deffksulhg5h7mbp1/vfs6uklvdce8qorira5eh5e9gflvvdst/1674453225000/17049954329865554887/*/1U1DN4XnhZmEaY7ivoPdj8tY0tau7X0sC?e=download&uuid=a55b9e5a-420d-46f1-a0ae-537150749976 [following]
--2023-01-23 05:54:47--  https://doc-00-2s-docs.googleusercontent.com/docs/securesc/ha0ro937gcuc7l7deffksulhg5h7mbp1/vfs6uklvdce8qorira5eh5e9gflvvdst/1674453225000/17049954329865554887/*/1U1DN4XnhZmEaY7ivoPdj8tY0tau7X0sC?e=download&uuid=a55b9e5a-420d-46f1-a0ae-537150749976
Resolving doc-00-2s-docs.googleusercontent.com (doc-00-2s-docs.googleusercontent.com)... 142.251.2.132, 2607:f8b0:4023:

In [33]:
train_set_benchmark = pd.read_csv('train_set_benchmark.csv', delimiter = '\t',header = None)
test_set_benchmark = pd.read_csv('test_set_benchmark.csv', delimiter = '\t',header = None)

In [36]:
print(f"Number of positive points in train set: {len(train_set_benchmark[train_set_benchmark[2] == 1])}")
print(f"Number of positive points in train set: {len(train_set_benchmark[train_set_benchmark[2] == 0])}")
print(f"Number of positive points in test set: {len(test_set_benchmark[test_set_benchmark[2] == 1])}")
print(f"Number of positive points in test set: {len(test_set_benchmark[test_set_benchmark[2] == 0])}")

Number of positive points in train set: 50000
Number of positive points in train set: 50000
Number of positive points in test set: 3415
Number of positive points in test set: 3415


In [35]:
train_set_benchmark.head(1)

0       1  2
0  8812  165140  1

## 2. Information about Pickle Files

---



In [39]:

!wget --no-check-certificate 'https://docs.google.com/uc?export=download&id=1-5bhZvskkFILsDhoNaaVlieUkkRdE3ir' -O our_dataset_negatome_pro_to_seq.pickle
!wget --no-check-certificate 'https://docs.google.com/uc?export=download&id=1Eu7OkO4WuVqaM4NqD6zF8r-Bbq1B5W7L' -O our_dataset_positive_pro_to_seq.pickle
!wget --no-check-certificate 'https://docs.google.com/uc?export=download&id=1k-BZJFS6u1UabFyLf32IkcigExSXkMYw' -O benchmark_pro_to_seq.pickle
!wget --no-check-certificate 'https://docs.google.com/uc?export=download&id=1kjHUjV5aIhaDADjV6w9__MtHteqWurQN' -O benchmark_sum_of_amino_acids_vectors.pickle
!wget --no-check-certificate 'https://docs.google.com/uc?export=download&id=1WpxAnS8-k3e-th5CumOpG8su5wL1wjp0' -O our_dataset_sum_of_amino_acids_vectors.pickle

--2023-01-23 06:06:05--  https://docs.google.com/uc?export=download&id=1-5bhZvskkFILsDhoNaaVlieUkkRdE3ir
Resolving docs.google.com (docs.google.com)... 142.251.2.113, 142.251.2.102, 142.251.2.101, ...
Connecting to docs.google.com (docs.google.com)|142.251.2.113|:443... connected.
HTTP request sent, awaiting response... 303 See Other
Location: https://doc-08-2s-docs.googleusercontent.com/docs/securesc/ha0ro937gcuc7l7deffksulhg5h7mbp1/u6653rs7r9mjqosks2dvrj9257qpa60d/1674453900000/17049954329865554887/*/1-5bhZvskkFILsDhoNaaVlieUkkRdE3ir?e=download&uuid=4899355a-54e3-41a9-bbc1-0b9b55db9c95 [following]
--2023-01-23 06:06:06--  https://doc-08-2s-docs.googleusercontent.com/docs/securesc/ha0ro937gcuc7l7deffksulhg5h7mbp1/u6653rs7r9mjqosks2dvrj9257qpa60d/1674453900000/17049954329865554887/*/1-5bhZvskkFILsDhoNaaVlieUkkRdE3ir?e=download&uuid=4899355a-54e3-41a9-bbc1-0b9b55db9c95
Resolving doc-08-2s-docs.googleusercontent.com (doc-08-2s-docs.googleusercontent.com)... 142.251.2.132, 2607:f8b0:4023:

In [38]:
import pickle

### 2.1 our_dataset_negatome_pro_to_seq.pickle

<pre>
It converts proteins of our dataset to sequence(only proteins from negative dateset)
</pre>

In [40]:
with open("our_dataset_negatome_pro_to_seq.pickle",'rb') as handle:
    our_dataset_negatome_pro_to_seq = pickle.load(handle)

In [42]:
our_dataset_negatome_pro_to_seq['PRO_8117']

'M S L K S E R R G I H V D Q S D L L C K K G C G Y Y G N P A W Q G F C S K C W R E E Y H K A R Q K Q I Q E D W E L A E R L Q R E E E E A F A S S Q S S Q G A Q S L T F S K F E E K K T N E K T R K V T T V K K F F S A S S R V G S K K E I Q E A K A P S P S I N R Q T S I E T D R V S K E F I E F L K T F H K T G Q E I Y K Q T K L F L E G M H Y K R D L S I E E Q S E C A Q D F Y H N V A E R M Q T R G K V P P E R V E K I M D Q I E K Y I M T R L Y K Y V F C P E T T D D E K K D L A I Q K R I R A L R W V T P Q M L C V P V N E D I P E V S D M V V K A I T D I I E M D S K R V P R D K L A C I T K C S K H I F N A I K I T K N E P A S A D D F L P T L I Y I V L K G N P P R L Q S N I Q Y I T R F C N P S R L M T G E D G Y Y F T N L C C A V A F I E K L D A Q S L N L S Q E D F D R Y M S G Q T S P R K Q E A E S W S P D A C L G V K Q M Y K N L D L L S Q L N E R Q E R I M N E A K K L E K D L I D W T D G I A R E V Q D I V E K Y P L E I K P P N Q P L A A I D S E N V E N D K L P P P L Q P Q V Y A G'

### 2.2 our_dataset_positive_pro_to_seq.pickle

<pre>
It converts proteins of our dataset to sequence(only proteins from positive dateset)
</pre>

In [43]:
with open("our_dataset_positive_pro_to_seq.pickle",'rb') as handle:
    our_dataset_positive_pro_to_seq = pickle.load(handle)



In [44]:
our_dataset_positive_pro_to_seq['PRO_1']

'M E P L N R T E V S E F F L K G F S G Y P A L E H L L F P L C S A M Y L V T L L G N T A I M A V S V L D I H L H T P V Y F F L G N L S T L D I C Y T P T F V P L M L V H L L S S R K T I S F A V C A I Q M C L S L S T G S T E C L L L A I T A Y D R Y L A I C Q P L R Y H V L M S H R L C V L L M G A A W V L C L L K S V T E M V I S M R L P F C G H H V V S H F T C K I L A V L K L A C G N T S V S E D F L L A G S I L L L P V P L A F I C L S Y L L I L A T I L R V P S A A R C C K A F S T C L A H L A V V L L F Y G T I I F M Y L K P K S K E A H I S D E V F T V L Y A M V T T M L N P T I Y S L R N K E V K E A A R K V W G R S R A S R'

In [48]:
x = len(our_dataset_positive_pro_to_seq) + len(our_dataset_negatome_pro_to_seq)
print(f"Total unique proteins on our dataset: {x}")

Total unique proteins on our dataset: 24781


### 2.3 benchmark_pro_to_seq.pickle

<pre>
It converts proteins of benchmark dataset to sequence

 benchmark_pro_to_seq.pickle
benchmark_sum_of_amino_acids_vectors.pickle
 our_dataset_sum_of_amino_acids_vectors.pickle
</pre>

In [50]:
with open("benchmark_pro_to_seq.pickle",'rb') as handle:
     benchmark_pro_to_seq = pickle.load(handle)
print(len( benchmark_pro_to_seq))

16640


### 2.4 benchmark_sum_of_amino_acids_vectors.pickle

<pre>
It converts proteins of benchmark protein's sequence to embedding vectors(input of siamese network)

</pre>

In [51]:
with open("benchmark_sum_of_amino_acids_vectors.pickle",'rb') as handle:
     benchmark_sum_of_amino_acids_vectors = pickle.load(handle)
print(len( benchmark_sum_of_amino_acids_vectors))

16640


### 2.5 our_dataset_sum_of_amino_acids_vectors.pickle

<pre>
It converts proteins of our dataset protein's sequence to embedding vectors(input of siamese network)
</pre>

In [60]:
with open("our_dataset_sum_of_amino_acids_vectors.pickle",'rb') as handle:
      our_dataset_sum_of_amino_acids_vectors = pickle.load(handle)
print(len( our_dataset_sum_of_amino_acids_vectors))

24016


### 2.6 amino_acids_embedded_vectors_bert_bfd.pickle(for our dataset)

<pre>

Embeddings of each amino acids from Protein Bert

<b>Path:</b> /mimer/NOBACKUP/groups/snic2019-35-21/PPI/pickle_files/amino_acids_embedded_vectors_bert_bfd.pickle

### 2.7 amino_acids_embedded_vectors_bert_albert.pickle(for our dataset)

<pre>

Embeddings of each amino acids from Protein Albert

<b>Path:</b> /mimer/NOBACKUP/groups/snic2019-35-21/PPI/pickle_files/amino_acids_embedded_vectors_bert_albert.pickle


</pre>

### 2.8 amino_acids_benchmark_embedded_vectors_bert_bfd.pickle(for benchmark)

<pre>

Embeddings of each amino acids from Protein Albert

<b>Path:</b> /mimer/NOBACKUP/groups/snic2019-35-21/PPI/pickle_files/amino_acids_benchmark_embedded_vectors_bert_bfd.pickle


</pre>

## Summary of dataset
#### Our dataset
<pre>
Number of unique proteins in our dataset: 24781
Max_length of proteins: 1024
Number of positive in points train set: 241819
Number of positive in points train set: 241787
</pre>

#### Benchmark dataset
<pre>
Number of unique proteins in benchmark dataset: 16640
Max_length of proteins: 1024
Number of positive in points train set: 50000
Number of positive in points train set: 50000
</pre>

## 3. Trained_model_weights and architecture




#### 3.1 Model Weights

In [55]:
!wget --load-cookies /tmp/cookies.txt "https://docs.google.com/uc?export=download&confirm=$(wget --quiet --save-cookies /tmp/cookies.txt --keep-session-cookies --no-check-certificate 'https://docs.google.com/uc?export=download&id=1NpFxtOpYr0cb8zB8x4Xah2mzY0lhQ1t7' -O- | sed -rn 's/.*confirm=([0-9A-Za-z_]+).*/\1\n/p')&id=1NpFxtOpYr0cb8zB8x4Xah2mzY0lhQ1t7" -O best_model_trained_our_dataset.pth && rm -rf /tmp/cookies.txt

--2023-01-23 06:24:17--  https://docs.google.com/uc?export=download&confirm=t&id=1NpFxtOpYr0cb8zB8x4Xah2mzY0lhQ1t7
Resolving docs.google.com (docs.google.com)... 142.250.101.113, 142.250.101.138, 142.250.101.102, ...
Connecting to docs.google.com (docs.google.com)|142.250.101.113|:443... connected.
HTTP request sent, awaiting response... 303 See Other
Location: https://doc-14-2s-docs.googleusercontent.com/docs/securesc/ha0ro937gcuc7l7deffksulhg5h7mbp1/1l6l0bccoj648o6687n8h2i467o2kitu/1674455025000/17049954329865554887/*/1NpFxtOpYr0cb8zB8x4Xah2mzY0lhQ1t7?e=download&uuid=8947cdd3-838c-4c5a-9991-46c50b4eee19 [following]
--2023-01-23 06:24:18--  https://doc-14-2s-docs.googleusercontent.com/docs/securesc/ha0ro937gcuc7l7deffksulhg5h7mbp1/1l6l0bccoj648o6687n8h2i467o2kitu/1674455025000/17049954329865554887/*/1NpFxtOpYr0cb8zB8x4Xah2mzY0lhQ1t7?e=download&uuid=8947cdd3-838c-4c5a-9991-46c50b4eee19
Resolving doc-14-2s-docs.googleusercontent.com (doc-14-2s-docs.googleusercontent.com)... 142.251.2.13

In [56]:
!wget --load-cookies /tmp/cookies.txt "https://docs.google.com/uc?export=download&confirm=$(wget --quiet --save-cookies /tmp/cookies.txt --keep-session-cookies --no-check-certificate 'https://docs.google.com/uc?export=download&id=1yPrF55izZwhl7oD-1oHXuHS2UO9s4stU' -O- | sed -rn 's/.*confirm=([0-9A-Za-z_]+).*/\1\n/p')&id=1yPrF55izZwhl7oD-1oHXuHS2UO9s4stU" -O best_model_trained_on_benchmark.pth && rm -rf /tmp/cookies.txt

--2023-01-23 06:26:01--  https://docs.google.com/uc?export=download&confirm=t&id=1yPrF55izZwhl7oD-1oHXuHS2UO9s4stU
Resolving docs.google.com (docs.google.com)... 142.250.101.139, 142.250.101.102, 142.250.101.113, ...
Connecting to docs.google.com (docs.google.com)|142.250.101.139|:443... connected.
HTTP request sent, awaiting response... 303 See Other
Location: https://doc-0k-2s-docs.googleusercontent.com/docs/securesc/ha0ro937gcuc7l7deffksulhg5h7mbp1/93td38pn1f2p3niei6btat2pe55jshhf/1674455100000/17049954329865554887/*/1yPrF55izZwhl7oD-1oHXuHS2UO9s4stU?e=download&uuid=187fc8d6-dc10-4274-8591-d9f3b291ff85 [following]
--2023-01-23 06:26:02--  https://doc-0k-2s-docs.googleusercontent.com/docs/securesc/ha0ro937gcuc7l7deffksulhg5h7mbp1/93td38pn1f2p3niei6btat2pe55jshhf/1674455100000/17049954329865554887/*/1yPrF55izZwhl7oD-1oHXuHS2UO9s4stU?e=download&uuid=187fc8d6-dc10-4274-8591-d9f3b291ff85
Resolving doc-0k-2s-docs.googleusercontent.com (doc-0k-2s-docs.googleusercontent.com)... 142.251.2.13

<pre>
<b>best_model_trained_our_dataset.pth</b>: Weights of model trained on our dataset (Accuracy: 80%)
<b>best_model_trained_on_benchmark.pth</b>: Weights of model trained on benchmark dataset (Accuracy: ~76%)
</pre>

#### 3.2 Model Architecture code

In [59]:
## Download model testing jupyter notebook

!wget https://raw.githubusercontent.com/danish2562022/PPI_prediction/main/model_results_analysis/model_testing.ipynb

--2023-01-23 06:34:03--  https://raw.githubusercontent.com/danish2562022/PPI_prediction/main/model_results_analysis/model_testing.ipynb
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 185.199.108.133, 185.199.109.133, 185.199.110.133, ...
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|185.199.108.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 21505 (21K) [text/plain]
Saving to: ‘model_testing.ipynb.2’

model_testing.ipynb 100%[===================>]  21.00K  --.-KB/s    in 0s      

2023-01-23 06:34:03 (56.3 MB/s) - ‘model_testing.ipynb.2’ saved [21505/21505]

